reference/tools/cmake/cmaketoolchain.rst

```diff
@@ -625,6 +625,26 @@ CMakeToolchain is affected by these ``[conf]`` variables:
 - **tools.cmake.cmaketoolchain:system_name** is not necessary in most cases and is only used to force-define ``CMAKE_SYSTEM_NAME``.
 - **tools.cmake.cmaketoolchain:system_version** is not necessary in most cases and is only used to force-define ``CMAKE_SYSTEM_VERSION``.
 - **tools.cmake.cmaketoolchain:system_processor** is not necessary in most cases and is only used to force-define ``CMAKE_SYSTEM_PROCESSOR``.
+- **tools.cmake.cmaketoolchain:extra_variables**: dict-like python object which specifies the CMake variable name and value. Value can be a plain string, number or a dict-like python object which must specifies ``value`` (string/number) , ``cache`` (boolean), ``type`` (CMake cache type) and optionally, ``docstring`` (string: defaulted to variable name). Potential override of CMakeToolchain defined variables, users are at their own risk. E.g.
+    
+.. code-block:: text
+
+    [conf]
+    tools.cmake.cmaketoolchain:extra_variables*={'STR': 'string value', 'NUM': 1.3, 'CMAKE_GENERATOR_INSTANCE': '${ENV}/buildTools/', 'SOME_DICT': {'value': '42 sense', 'cache': True, 'type': 'BOOL', 'docstring': 'test cache var'}}
+
+Resulting in:
+
+.. code-block:: cmake
+
+    set(STR "string value")
+    set(NUM 1.3)
+    set(CMAKE_GENERATOR_INSTANCE "${ENV}/buildTools/")
+    set(SOME_DICT "42 sense" CACHE BOOL "test cache var")
+
+.. tip::
+
+    Use the ``*=`` operator described in :ref:`configuration data operators <configuration_data_operators>` to update the dictionary merging it with previous definitions of ``extra_variables``.
```

Comment:
I am not sure if this tip is relevant here. Is there a good example of a use case of this usage for extra_variables?

Anyway, I think it reads better this way. WDYT?
```suggestion
    Use the :ref:`configuration data operator<configuration_data_operators>` ``*=`` to **update** (instead of redefine) conf variables already defined in profiles or global configuration.
```

Reply:
Okey thanks!